Public/Src/Engine/Scheduler/Graph/GraphFragmentBuilder.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Linq;
+using System.Threading;
+using BuildXL.Ipc;
+using BuildXL.Ipc.Interfaces;
+using BuildXL.Pips;
+using BuildXL.Pips.Builders;
+using BuildXL.Pips.Operations;
+using BuildXL.Utilities;
+using BuildXL.Utilities.Configuration;
+using BuildXL.Utilities.Instrumentation.Common;
+using JetBrains.Annotations;
+using static BuildXL.Scheduler.Graph.PipGraph;
+
+namespace BuildXL.Scheduler.Graph
+{
+    /// <summary>
+    /// Class for building graph fragments.
+    /// </summary>
+    public class GraphFragmentBuilder : IPipGraph

```

Comment:
Could we mvoe this to BuildXL.Pips assembly?

This should be general use and not have to tie fully into the schedule which almost pulls in the entire engine

Reply:
It has the functionality as a graph builder. In the next stage, we're going to infer some dependencies to make graph deserialization parallel.

---
In reply to: [313653006](https://github.com/microsoft/BuildXL/pull/730#discussion_r313653006) [](ancestors = 313653006)